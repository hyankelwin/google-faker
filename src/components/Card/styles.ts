import styled from 'styled-components';
import Color from '../../styles/color';

export const Container = styled.div`
  border: 1px solid ${Color.grey[200]};
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: start;
  max-width: 400px;
  border-radius: 4px;
`;

export const Caption = styled.p`
  font-size: 12px;
  padding-top: 10px;
`;
export const Title = styled.h3`
  padding-top: 5px;
  padding-bottom: 5px;
`;
export const Description = styled.p`
  font-size: 14px;
`;

export const Image = styled.img`
  width: 100%;
`;
