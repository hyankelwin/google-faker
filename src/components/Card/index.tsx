import React from 'react';
import { Animal } from '../../api/google-faker-api/search/interfaces/animal.interface';
import { Container, Image, Description, Caption, Title } from './styles';

interface CardProps {
  item: Animal;
}

const Card: React.FC<CardProps> = ({ item }) => (
  <Container>
    <Image src={item.image} />
    <Caption>{item.site}</Caption>
    <Title>{item.name}</Title>
    <Description>{item.description}</Description>
  </Container>
);

export default Card;
