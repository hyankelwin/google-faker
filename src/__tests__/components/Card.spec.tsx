import { mount } from 'enzyme';
import Card from '../../components/Card';
import { mockAnimal } from '../../__mocks__/animal.mock';

describe('Card component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render Card component with success', () => {
      const wrapper = mount(<Card item={mockAnimal} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
