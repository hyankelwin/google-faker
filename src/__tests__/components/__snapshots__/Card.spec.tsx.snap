// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should be able to passed in all tests on index.tsx should be able to render Card component with success 1`] = `
"<Card item={{...}}>
  <styled.div>
    <div className=\\"sc-aXZVg hwnWVx\\">
      <styled.img src=\\"https://inovaveterinaria.com.br/wp-content/uploads/2016/01/Gato-siames-INOVA-scaled.jpg\\">
        <img src=\\"https://inovaveterinaria.com.br/wp-content/uploads/2016/01/Gato-siames-INOVA-scaled.jpg\\" className=\\"sc-dcJsrY kLFbiy\\" />
      </styled.img>
      <styled.p>
        <p className=\\"sc-gEvEer cAUUCC\\">
          http://google.com.br
        </p>
      </styled.p>
      <styled.h3>
        <h3 className=\\"sc-eqUAAy kVxYHK\\">
          SiamÃªs
        </h3>
      </styled.h3>
      <styled.p>
        <p className=\\"sc-fqkvVR cdqkYV\\">
          This cat is the most beautiful than all cats
        </p>
      </styled.p>
    </div>
  </styled.div>
</Card>"
`;
